cli/azd/cmd/container.go

```diff
@@ -123,6 +125,65 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		return writer
 	})
 
+	container.RegisterSingleton(func() *envFlag {
+		// Get the current cmd flags for the executing command
+		var currentFlags flags
+		err := container.Resolve(&currentFlags)
```

Comment:
Some questions just out of curiosity:

1. Would this work if we simply did `container.Resolve(flagsWithEnv)`?
1. Would this work if we relied on the DI declaration, i.e. `func(envFlag flagsWithEnv)`?

Reply:
Unfortunately no.  Since all flags implement the `flags` interface this is a safe resolution of the current command flags.  Then we need to check if the flags also implement the `flagsWithEnv`.  If so, we have the flags otherwise determine their aren't any env flags.